src/ripple/rpc/impl/TransactionSign.h

```diff
@@ -135,16 +71,24 @@ Json::Value transactionSign (
     NetworkOPs& netOPs,
     Role role)
 {
-    detail::TxnSignApiFacade apiFacade (netOPs);
-    return transactionSign (params, failType, apiFacade, role);
+    return transactionSign (
+        params,
+        failType,
+        role,
+        getApp().getLedgerMaster().getValidatedLedgerAge(),
+        getApp().getFeeTrack(),
+        getApp().getLedgerMaster().getCurrentLedger());
 }
 
 /** Returns a Json::objectValue. */
 Json::Value transactionSubmit (
     Json::Value params,  // Passed by value so it can be modified locally.
     NetworkOPs::FailHard failType,
-    detail::TxnSignApiFacade& apiFacade,
-    Role role);
+    Role role,
+    int validatedLedgerAge,
+    LoadFeeTrack const& feeTrack,
+    std::shared_ptr<ReadView const> ledger,
+    detail::ProcessTransactionFn const& processTransaction);
```

Comment:
That's fair.  I believe you're arguing that the non-inlined interfaces for `transactionSign()`, `transactionSubmit()`, `transactionSignFor()`, and `transactionSubmitMultiSigned()` should be moved into the `detail` namespace.  I think that's a great idea.  The non-inlined forms are only exposed for unit testing purposes.